design-documents/graph-ql/coverage/url-resolver-identifier.md

```diff
@@ -0,0 +1,68 @@
+ # UrlResolver identifier 
+ ## Problem statement   
+ 
+In the client flow, for PWA, in order to load a entity/template, one must make a query to the `urlResolver` to obtain the *id* and *Entity Type* of the content that can be Product, Category or CMS.
+
+####Example:
+#####Request:
+```graphql
+urlResolver(url: "/breathe-easy-tank") {
+  id
+  type
+}
+```
+#####Response:
+```json
+{
+  "data": {
+    "urlResolver": {
+      "id": 1820,
+      "type": "PRODUCT"
+    }
+  }
+}
+```
+
+Currently we can't use this Id in Product to filter by, because of how Search API works, also CMS has another unique identifier that can be used.
+Also we want not to expose database autoincrement ID in graphql in the future because of replacements with UUID.
+
+ ## Proposed solution
+
+Deprecate `id` field from `urlResolver` and introduce a new field called identifier, unique to each entity that's not the Id from the database.
+
+```graphql
+urlResolver(url: String) {
```

Comment:
We may need to keep `id` and introduce `urlResolverV2` for client-side caching. We are going to have a discussion today with domain architects responsible for affected areas.

Reply:
added a comment at the bottom part of alternatives

Reply:
Here are the final list of steps proposed (based on architectural discussion):
1. introduce “ids in” to products filter, type ids:[ID!]
1.  introduce urlResolverV2, whith return type of id:ID
     1.  id is id of the entity, casted to string
     1.  if only ids are passed, bypass Search API and fetch data directly from DB
     1. if ids + other filters are passed, consider this an unsupported case. To be implemented later (create a task). Throw an exception in case of such query.

Reply:
@cpartica in the adjusted proposal I see product filter by "id", not by "ids". Is this a mistake or there is a reason to avoid "ids"?

The following guidelines should be added to the proposal in some form, they are really important:

```
- if only ids are passed, bypass Search API and fetch data directly from DB
- if ids + other filters are passed, consider this an unsupported case. To be implemented later (create a task). Throw an exception in case of such query.
```

Reply:
added those lines